include/zephyr/net/icmp.h

```diff
@@ -191,6 +191,30 @@ int net_icmp_send_echo_request(struct net_icmp_ctx *ctx,
 			       struct net_icmp_ping_params *params,
 			       void *user_data);
 
+/**
+ * @brief Send ICMP echo request message from k_work handler.
+ *
+ * @details This function can be used to send ICMP echo request from
+ *          a k_work handler. This variant will not wait if the network
+ *          packet allocation fails during sending. This will avoid a warning
+ *          message in the log about timeout setting set to K_NO_WAIT.
+ *
+ * @param ctx ICMP context used in this request.
+ * @param iface Network interface, can be set to NULL in which case the
+ *        interface is selected according to destination address.
+ * @param dst IP address of the target host.
+ * @param params Echo-Request specific parameters. May be NULL in which case
+ *        suitable default parameters are used.
+ * @param user_data User supplied opaque data passed to the handler. May be NULL.
+ *
+ * @return Return 0 if the sending succeed, <0 otherwise.
+ */
+int net_icmp_send_echo_request_from_work(struct net_icmp_ctx *ctx,
```

Comment:
Personally I'm a bit puzzled with the function name, it describes a possible use-case for the function, and not the function behavior, I'd prefer something like `net_icmp_send_echo_request_no_wait()`. But no strong push though.

Reply:
I was pondering with the name too, the `no_wait` variant gave a hint to me that there is no wait for echo-request command itself. But if it is clear what it indicates, we can certainly rename it.

Reply:
After second thought, I will change this to `no_wait` which I had originally but then started to overthink the name too much.